postcss.config.mjs

```diff
@@ -0,0 +1,86 @@
+import autoprefixer from 'autoprefixer'
+import cssnano from 'cssnano'
+import cssnanoPresetDefault from 'cssnano-preset-default'
+import { minimatch } from 'minimatch'
+import postcss from 'postcss'
+import pseudoclasses from 'postcss-pseudo-classes'
+import scss from 'postcss-scss'
+import unmq from 'postcss-unmq'
+import unopacity from 'postcss-unopacity'
+import unrgba from 'postcss-unrgba'
+
+import { pkg } from './config/index.js'
+import { isDev } from './tasks/task-arguments.mjs'
+
+/**
+ * PostCSS config
+ *
+ * @param {import('postcss-load-config').ConfigContext} ctx - PostCSS context
+ * @returns {import('postcss-load-config').Config} PostCSS config
+ */
+export default ({ from = '', to = '', env = 'production' }) => {
+  const plugins = []
+
+  // Browserslist IE8 environment
+  if (from.includes('-ie8')) {
+    env = 'oldie'
+  }
+
+  // Add vendor prefixes
+  plugins.push(autoprefixer({ env }))
+
+  // Add review app auto-generated 'companion' classes for each pseudo-class
+  // For example ':hover' and ':focus' classes to simulate form label states
+  if (minimatch(from, '**/app/stylesheets/*')) {
+    plugins.push(pseudoclasses({
+      allCombinations: true,
+      restrictTo: [
+        ':link',
+        ':visited',
+        ':hover',
+        ':active',
+        ':focus'
+      ]
+    }))
+  }
+
+  // Transpile CSS for Internet Explorer
+  if (env === 'oldie') {
+    plugins.push(
+      unmq(),
+      unopacity({ browsers: 'ie 8' }),
+      unrgba({ filter: true })
+    )
+  }
+
+  // Add GOV.UK Frontend release version
+  if (!isDev) {
+    plugins.push({
+      postcssPlugin: 'govuk-frontend-version',
+      Declaration: {
+        '--govuk-frontend-version': (decl) => {
+          decl.value = `"${pkg.version}"`
+        }
+      }
```

Comment:
Is there a short explanation we could add, or some postdoc docs we could link to, to help understand what this is doing?

Reply:
Added the following comments with linked I used to find this approach:

```mjs
// Find CSS declaration for version, update value
// https://github.com/postcss/postcss/blob/main/docs/writing-a-plugin.md
// https://postcss.org/api/#declaration
```